crates/roc_std/src/roc_dict.rs

```diff
@@ -132,6 +132,7 @@ impl<K: Debug, V: Debug> Debug for RocDict<K, V> {
 /// contiguously in memory. If we separate them at some point, we'll need to
 /// change this implementation drastically!
 #[derive(Eq)]
+#[repr(C)]
 union RocDictItem<K, V> {
```

Comment:
A note: this is the old definition for RocDict when it was just a list. If you try to run any tests with this, it will fail. We need eventually update roc_dict to reflect the fact we use a real hashmap now.

A good stop gap would probably be to remove all of the methods on roc_dict and just make the type right so it can be passed around. But that shouldn't neeed to be part of this pr.

Reply:
can we rip out RocDict from std and have glue generate it instead? Otherwise we will need a bunch of duplicate work regardless to handle how Roc does alignment and all the unwrapping optimizations

Reply:
That is the long term plan. Glue should generate all of `roc_std`. That said, RocDict still has a big annoyance. Likely glue will want to generate a type that is usable to the platform. This requires knowing the internal details of RocDict and generating functions that can actually use the dictionary. That said, I guess that RocDict could also be opaque to the platform. Then they could just expose a roc function for updating/reading the dictionary if they need to. That way, roc would generate all interaction with dictionaries. Lose some performance and inlining, but would mean that this can be fully decoupled.